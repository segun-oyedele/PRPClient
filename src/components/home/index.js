export * from './HomeCardContainer';
export * from './HomeCardItem';