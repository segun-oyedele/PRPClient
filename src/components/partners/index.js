export * from './SearchBar';
export * from './TableBody';