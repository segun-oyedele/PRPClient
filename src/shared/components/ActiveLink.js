import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import { useEffect, useState } from 'react';

export const ActiveLink = ({ label, href, stylesClass, navPath }) => {
  const router = useRouter();

  const [isActive, setIsActive] = useState(router.pathname === navPath);
  const activeStyles = isActive ? 'md:text-white raleway-eb cursor-default' : 'nav_link-inactive';

  const handleClick = (event) => {
    event.preventDefault();
    setIsActive(true);
    router.push(href);
  }

  useEffect(() => {
    setIsActive(router.pathname === navPath);
  }, [router.pathname]);
  

  if( isActive ) {
    return (
      <a
        className={`${ activeStyles } ${ stylesClass }`}
      >{ label }</a>
    );
  } else {
    return(
      <a
        href={ href }
        onClick={ handleClick }
        className={`${ activeStyles } ${ stylesClass }`}
      >{ label }</a>
    );
  }
};

ActiveLink.propTypes = {
  label: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  stylesClass: PropTypes.string,
  navPath: PropTypes.string.isRequired
};
