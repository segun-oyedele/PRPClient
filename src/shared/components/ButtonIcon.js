import PropTypes from 'prop-types';

const btnDefaultColors = 'bg-indigo-500 hover:bg-indigo-600 focus:bg-indigo-600 active:bg-indigo-600';
const labelDefaultColors = 'text-white';
const btnDefaultStyles = 'rounded-tr rounded-br';
const btnDefaultSpacing = 'px-4 py-2.5';

export const ButtonIcon = ({ children, onClick, btnColors, btnStyles, btnLabel, labelStyles, labelColors, buttonSpacing, iconOnRight }) => {
  return (
    <button
      className={`btn btn_icon focus:outline-none focus:ring-0 active:shadow-lg transition duration-300 ease-in-out flex items-center ${ btnStyles || btnDefaultStyles } ${ btnColors || btnDefaultColors } ${ buttonSpacing || btnDefaultSpacing }`}
      type="button"
      id="button-addon2"
      onClick={ () => onClick() }
    >
      { !iconOnRight && children }

      { btnLabel && <span className={`font-medium tracking-wide ${ labelColors || labelDefaultColors } ${ labelStyles || '' }`}>{ btnLabel }</span> }
    
      { iconOnRight && children }
    </button>
  );
};

ButtonIcon.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func.isRequired,
  btnColors: PropTypes.string,
  btnStyles: PropTypes.string,
  btnLabel: PropTypes.string,
  labelStyles: PropTypes.string,
  labelColors: PropTypes.string
};
