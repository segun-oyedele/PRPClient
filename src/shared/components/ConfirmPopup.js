import PropTypes from 'prop-types';
import { Fragment } from 'react';
import { Dialog, Transition } from '@headlessui/react';

export const ConfirmPopup = ({ isOpen, setIsOpen, onClick, title, description, colorsPrimaryBoton, buttonTitle, showButton = true }) => {

  const colorsBtn = colorsPrimaryBoton || "bg-indigo-500 border border-transparent rounded-md hover:bg-indigo-600 focus:bg-indigo-600 active:bg-indigo-600 text-white";

  function closeModal() {
    setIsOpen(false);
  }

  const handleClick = () => {
    closeModal();
    onClick();
  }

  return (
    <>
      <Transition appear show={isOpen} as={Fragment}>
        <Dialog as="div" className="relative z-50" onClose={closeModal}>
          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <div className="fixed inset-0 bg-black bg-opacity-25" />
          </Transition.Child>

          <div className="fixed inset-0 overflow-y-auto">
            <div className="flex items-center justify-center min-h-full p-4 text-center">
              <Transition.Child
                as={Fragment}
                enter="ease-out duration-300"
                enterFrom="opacity-0 scale-95"
                enterTo="opacity-100 scale-100"
                leave="ease-in duration-200"
                leaveFrom="opacity-100 scale-100"
                leaveTo="opacity-0 scale-95"
              >
                <Dialog.Panel className="w-full max-w-md p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-2xl">
                  <Dialog.Title
                    as="h3"
                    className="text-lg font-medium leading-6 text-gray-900"
                  >
                    { title }
                  </Dialog.Title>
                  <div className="mt-2">
                    <p className="text-sm text-gray-500">
                      { description }
                    </p>
                  </div>
                  { showButton &&
                    <div className="grid grid-cols-2 gap-2 mt-4">
                      <button
                        type="button"
                        className={`${ colorsBtn } inline-flex justify-center px-4 py-2 text-sm font-medium transition duration-300 border border-transparent rounded-md focus:outline-none focus-visible:ring-2`}
                        onClick={ handleClick }
                      >
                        { buttonTitle }
                      </button>
                      <button
                        type="button"
                        className="inline-flex justify-center px-4 py-2 text-sm font-medium text-indigo-600 transition duration-300 bg-white border border-transparent rounded-md hover:border-indigo-500 focus:border-indigo-500 active:border-indigo-500 focus:outline-none focus-visible:ring-2 "
                        onClick={ closeModal }
                      >
                        Cancel
                      </button>
                    </div>
                  }
                </Dialog.Panel>
              </Transition.Child>
            </div>
          </div>
        </Dialog>
      </Transition>
    </>
  );
};

ConfirmPopup.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  setIsOpen: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  colorsPrimaryBoton: PropTypes.string,
  buttonTitle: PropTypes.string.isRequired,
  showButton: PropTypes.bool
}