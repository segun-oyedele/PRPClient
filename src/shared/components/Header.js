import { Navbar } from './';
import Link from 'next/link';
import Image from 'next/image';
import { useRouter } from 'next/router';
import { navLinks } from '../data';
import { useEffect, useState } from 'react';

export const Header = () => {
  
  const router = useRouter();
  const [isActive, setIsActive] = useState(router.pathname === navLinks[0].navPath);

  useEffect(() => {
    setIsActive(router.pathname === navLinks[0].navPath);
  }, [router.pathname]);

  return (
    <header className="sticky top-0 z-10 md:top-auto md:static md:z-auto header-tag primary-blue">
      <div className="container flex items-center h-full min-w-full p-5 md:p-0 md:px-4">
        { isActive ?
          <a className="flex items-center gap-3 cursor-default">
            <Image
              alt='Last Mile Logo'
              className="w-16"
              width={106}
              height={50}
              src={`${process.env.iisPath}/img/last_mile_logo.png`}
              title="Last-Mile-Logo"
            />
          </a>
          :
          <Link href={`${navLinks[0].to}`}>
            <a className="flex items-center gap-3">
              <Image
                alt='Last Mile Logo'
                className="w-16"
                width={106}
                height={50}
                src={`${process.env.iisPath}/img/last_mile_logo.png`}
                title="Last-Mile-Logo"
              />
            </a>
          </Link>
        }

        <Navbar />
      </div>
    </header>
  );
};
