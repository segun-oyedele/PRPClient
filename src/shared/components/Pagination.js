import PropTypes from 'prop-types';

export const Pagination = ({ partnersPerPage, totalPartners, handleChangePage, currentPage }) => {

  const pageNumbers = [];
  const totalPages = Math.ceil( totalPartners / partnersPerPage );
  const prevAvailable = currentPage > 1;
  const nextAvailable = currentPage < totalPages;

  for (let i = 0; i < totalPages; i++) {
    pageNumbers.push(i+1);
  };

  return (
    <div className="grid items-center justify-center mt-5 gap-y-5 md:grid-flow-col md:justify-between justify-items-center pagination">

      <span className="current-pagination">Showing { currentPage * 10 > totalPartners ? totalPartners : currentPage * 10 } of { totalPartners }</span>

      {
        !nextAvailable && !prevAvailable ?
          null
        : 
        <div className="flex items-center gap-2 pagination_buttons">
          <button
            className="grid items-center grid-flow-col gap-1 px-4 py-2 text-base raleway-bl"
            onClick={ () => prevAvailable ? handleChangePage(currentPage - 1) : null }
          >
            <span>Previous</span>
          </button>
        {
          pageNumbers.map( number => (
            <button
              key={ number }
              className={`raleway-bl text-base w-6 h-6 leading-5 ${ currentPage === number ? 'current-page cursor-default bg-black text-white rounded-sm' : currentPage === (number - 1) ? 'bg-white next-page' : 'bg-white' } ${ (number > (currentPage + 2)) || (number < (currentPage - 2)) ? 'hidden' : ''  } pagination-item`}
              onClick={ () => handleChangePage(number) }
            >{ number }</button>
          ))
        }
        <button
          className="grid items-center grid-flow-col gap-1 px-4 py-2 text-base raleway-bl"
          onClick={ () => nextAvailable ? handleChangePage(currentPage + 1) : null }
        >
          <span>Next</span>
        </button>
        </div>
      }
    </div>
  );
};

Pagination.propTypes = {
  partnersPerPage: PropTypes.number.isRequired,
  totalPartners: PropTypes.number.isRequired,
  handleChangePage: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired
};
