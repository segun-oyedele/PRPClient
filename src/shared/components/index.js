export * from './ActiveLink';
export * from './BurgerIcon';
export * from './ButtonIcon';
export * from './Header';
export * from './Layouts';
export * from './Links';
export * from './Navbar';
export * from './PageTitle';
export * from './Pagination';
export * from './Sidebar';
export * from './Table';
export * from './TableHead';
