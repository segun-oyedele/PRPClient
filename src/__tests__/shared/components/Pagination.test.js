import { render, screen, fireEvent} from '@testing-library/react';
import { Pagination } from '@/shared/components/Pagination';

describe('Test on Pagination', () => {
  
  test('should match with snapshot', () => {
    const { container } = render(<Pagination handleChangePage={ jest.fn() } totalPartners={ 12 } partnersPerPage={ 2 } currentPage={ 1 } />);

    expect( container ).toMatchSnapshot();
  });
  
  test('next button should to call handleChangePage at least one time', () => {
    const handleChangePage = jest.fn();

    render(<Pagination handleChangePage={ handleChangePage } totalPartners={ 12 } partnersPerPage={ 2 } currentPage={ 1 } />);

    const button = screen.getByRole('button', {
      name: /next/i
    });
    fireEvent.click(button);

    expect( handleChangePage ).toHaveBeenCalled();
    expect( handleChangePage ).toHaveBeenCalledTimes(1);
  });
  
  test('next button should do not have to call handleChangePage', () => {
    const handleChangePage = jest.fn();

    render(<Pagination handleChangePage={ handleChangePage } totalPartners={ 12 } partnersPerPage={ 2 } currentPage={ Math.ceil( 12 / 2 ) } />);

    const button = screen.getByRole('button', {
      name: /next/i
    });
    fireEvent.click(button);

    expect( handleChangePage ).not.toHaveBeenCalled();
    expect( handleChangePage ).toHaveBeenCalledTimes(0);
  });
  
  test('prev button should to call handleChangePage at least one time', () => {
    const handleChangePage = jest.fn();

    render(<Pagination handleChangePage={ handleChangePage } totalPartners={ 12 } partnersPerPage={ 2 } currentPage={ 2 } />);

    const button = screen.getByRole('button', {
      name: /prev/i
    });
    fireEvent.click(button);

    expect( handleChangePage ).toHaveBeenCalled();
    expect( handleChangePage ).toHaveBeenCalledTimes(1);
  });
  
  test('prev button should do not have to call handleChangePage', () => {
    const handleChangePage = jest.fn();

    render(<Pagination handleChangePage={ handleChangePage } totalPartners={ 12 } partnersPerPage={ 2 } currentPage={ 1 } />);

    const button = screen.getByRole('button', {
      name: /prev/i
    });
    fireEvent.click(button);

    expect( handleChangePage ).not.toHaveBeenCalled();
    expect( handleChangePage ).toHaveBeenCalledTimes(0);
  });

})