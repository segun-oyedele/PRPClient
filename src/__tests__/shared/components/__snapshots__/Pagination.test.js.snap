// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Pagination should match with snapshot 1`] = `
<div>
  <div
    class="grid justify-center grid-flow-col gap-3 mt-10 justify-items-center pagination"
  >
    <button
      class="grid items-center grid-flow-col gap-1 px-4 py-2 rounded shadow-md bg-gray-100 text-gray-500 cursor-default"
    >
      <svg
        aria-hidden="true"
        class="w-4 h-4 text-gray-500"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M15.707 15.707a1 1 0 01-1.414 0l-5-5a1 1 0 010-1.414l5-5a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 010 1.414zm-6 0a1 1 0 01-1.414 0l-5-5a1 1 0 010-1.414l5-5a1 1 0 011.414 1.414L5.414 10l4.293 4.293a1 1 0 010 1.414z"
          fill-rule="evenodd"
        />
      </svg>
      <span>
        Prev
      </span>
    </button>
    <button
      class="current-page cursor-default bg-indigo-600 text-white  px-4 py-2 rounded shadow-md pagination-item"
    >
      1
    </button>
    <button
      class="bg-white next-page  px-4 py-2 rounded shadow-md pagination-item"
    >
      2
    </button>
    <button
      class="bg-white hidden px-4 py-2 rounded shadow-md pagination-item"
    >
      3
    </button>
    <button
      class="bg-white hidden px-4 py-2 rounded shadow-md pagination-item"
    >
      4
    </button>
    <button
      class="bg-white hidden px-4 py-2 rounded shadow-md pagination-item"
    >
      5
    </button>
    <button
      class="bg-white hidden px-4 py-2 rounded shadow-md pagination-item"
    >
      6
    </button>
    <button
      class="grid items-center grid-flow-col gap-1 px-4 py-2 rounded shadow-md bg-white"
    >
      <span>
        Next
      </span>
      <svg
        aria-hidden="true"
        class="w-4 h-4 text-gray-500"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.293 15.707a1 1 0 010-1.414L14.586 10l-4.293-4.293a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M4.293 15.707a1 1 0 010-1.414L8.586 10 4.293 5.707a1 1 0 011.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
